liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/PluginConfigSupport.java

```diff
@@ -191,4 +193,29 @@ protected String stripVersionFromName(String name, String version) {
             return name;
         }
     }
+    
+    protected void addAppConfiguration(String artifactId) throws Exception {
+   
+        log.warn(messages.getString("info.install.app.not.configured"));
+         
+        // Add webApplication configuration into the target server.xml. 
+        File serverXML = new File(serverDirectory, "server.xml");
+        ServerXmlDocument.addAppElment(serverXML, artifactId);
+        
+        log.info(MessageFormat.format(messages.getString("info.install.app.add.configuration"), artifactId));
+    }
+    
+    protected boolean isApplicationConfigured() throws Exception {
+        File serverXML = new File(serverDirectory, "server.xml");
+        return ServerXmlDocument.isFoundTagNames(serverXML.getCanonicalPath(), 	
+                                                 new String[] {"application", "webApplication"});
+    }
+    
+    protected String getAppsDirectory() {
```

Comment:
We need the getAppsDirectory to return the smart default not just for install-apps goal